src/osd/OSD.cc

```diff
@@ -6301,17 +6301,27 @@ int OSD::probe_smart_device(const char *device, int timeout, std::string *result
   ret = output.read_fd(smartctl.get_stdout(), 100*1024);
   if (ret < 0) {
     derr << "failed read from smartctl: " << cpp_strerror(-ret) << dendl;
-    return ret;
+    goto out;
```

Comment:
could you avoid using `goto` here by following the same pattern in CrushLocation.cc?

Reply:
@guzhongyan is it possible to move the join call out of the if checks? put in other words, you are calling `smartctl.join()` at multiple places. could you do the error handling in same way as we do in CrushLocation.cc ?

Reply:
@tchaikov got it. commit updated. please review.